diag_manager/fms_diag_bbox.F90

```diff
@@ -162,6 +228,97 @@ SUBROUTINE reset_bounds_from_array_5D(this, array)
       this%kmax = UBOUND(array,3)
    END SUBROUTINE  reset_bounds_from_array_5D
 
+  !> @brief Updates indices based on presence/absence of input indices is, js, ks, ie, je, and ke.
+  ! Computes halo sizes in the I and J dimensions.
+  ! This routine is intended to be used in diag manager.
+  !> @return .false. if there is no error else .true.
+  function recondition_indices(indices, field, is_in, js_in, ks_in, &
```

Comment:
this should just be a subroutine instead of a function.  The error checking is happening in the routine, not by the caller.